eslint.config.js

```diff
@@ -0,0 +1,120 @@
+import js from '@eslint/js';
+import globals from 'globals';
+import mdcs from 'eslint-config-mdcs';
+import compat from 'eslint-plugin-compat';
+import html from 'eslint-plugin-html';
+import importPlugin from 'eslint-plugin-import';
+import jsdoc from 'eslint-plugin-jsdoc';
+
+export default [
+	// files to ignore
+	{
+		name: 'files to ignore',
+		ignores: [
+			'**/node_modules/**',
+			'**/build/**',
+			'**/libs/**',
+			'**/ifc/**',
+			'**/prettify.js',
+			'**/prettify/**',
+			'**/3rdparty/**',
+			'**/shapefile.js',
+			'**/vendor/**',
+			'**/fuse/**'
+		]
+	},
+
+	// recommended
+	js.configs.recommended,
+	// jsdoc.configs['flat/recommended'],
+
+	// base rules
+	{
+		name: 'base rules',
+		files: [ '**/*.js', '**/*.html' ],
+		plugins: {
+			html,
+			import: importPlugin,
+			compat,
+			jsdoc
+		},
+		languageOptions: {
+			ecmaVersion: 2018,
+			sourceType: 'module',
+			globals: {
+				...globals.browser,
+				...globals.node,
+				...globals.es2017,
+				__THREE_DEVTOOLS__: 'readonly',
+				potpack: 'readonly',
+				fflate: 'readonly',
+				Stats: 'readonly',
+				XRWebGLBinding: 'readonly',
+				XRWebGLLayer: 'readonly',
+				GPUShaderStage: 'readonly',
+				GPUBufferUsage: 'readonly',
+				GPUTextureUsage: 'readonly',
+				GPUTexture: 'readonly',
+				GPUMapMode: 'readonly',
+				QUnit: 'readonly',
+				Ammo: 'readonly',
+				XRRigidTransform: 'readonly',
+				XRMediaBinding: 'readonly',
+				CodeMirror: 'readonly',
+				esprima: 'readonly',
+				jsonlint: 'readonly',
+				VideoFrame: 'readonly',
+				VideoDecoder: 'readonly',
+				Float16Array: 'readonly',
+				BigInt: 'readonly',
+				BigUint64Array: 'readonly'
+			},
+		},
+		settings: {
+			polyfills: [
+				'WebGL2RenderingContext'
+			],
+			jsdoc: {
+				preferredTypes: {
+					Any: 'any',
+					Boolean: 'boolean',
+					Number: 'number',
+					object: 'Object',
+					String: 'string'
+				},
+				tagNamePreference: {
+					returns: 'return',
+					augments: 'augments',
+					extends: 'augments'
+				},
+			},
+		},
+		rules: {
+			...mdcs.rules,
+			...compat.configs.recommended.rules,
+			'no-throw-literal': 'error',
+			'quotes': [ 'error', 'single' ],
+			'prefer-const': [ 'error', {
+				destructuring: 'any',
+				ignoreReadBeforeAssign: false
+			} ],
+			'no-irregular-whitespace': 'error',
+			'no-duplicate-imports': 'error',
+			'prefer-spread': 'error',
+
+			'no-useless-escape': 'off',
+			'no-case-declarations': 'off',
+			'no-cond-assign': 'off',
+			'getter-return': 'off',
+			'no-async-promise-executor': 'off',
+			'no-empty': 'off',
+
+			// 'jsdoc/check-types': 'error',
```

Comment:
When this is enabled there are some errors in the JSDoc comments that are caught. I'll leave it to another contribution to re-enable and fix these. It may be worth enabling the JSDoc "recommended" settings above to see if there are any rules worth keeping from there, too.